packages/ag-charts-community/src/chart/overlay/chartOverlays.ts

```diff
@@ -26,11 +32,55 @@ export class ChartOverlays {
         }
     }
 
+    loading: Overlay;
     noData: Overlay;
     noVisibleSeries: Overlay;
 
     public destroy() {
+        this.loading.hide();
         this.noData.hide();
         this.noVisibleSeries.hide();
     }
+
+    private renderLoadingSpinner(parent: HTMLElement, animationManager: AnimationManager) {
+        const container = this.createElement(parent, 'div');
+        container.style.alignItems = 'center';
+        container.style.boxSizing = 'border-box';
+        container.style.display = 'flex';
+        container.style.flexDirection = 'column';
+        container.style.justifyContent = 'center';
+        container.style.margin = '8px';
+        container.style.height = '100%';
+        container.style.font = '12px Verdana, sans-serif';
+        container.style.animation = `ag-charts-loading ${
+            ADD_PHASE.animationDuration * animationManager.defaultDuration
+        }ms linear ${ADD_PHASE.animationDelay * animationManager.defaultDuration}ms both`;
+
+        const spinner = this.createElement(container, 'div');
+        spinner.style.width = '10px';
+        spinner.style.height = '10px';
+        spinner.style.borderRadius = '50%';
+        spinner.style.borderWidth = '2px';
+        spinner.style.borderStyle = 'solid';
+        spinner.style.borderColor = 'rgb(0, 0, 0) rgba(0, 0, 0, 0)';
+        spinner.style.animation = 'ag-charts-loading-spinner 1s infinite';
+
+        const label = this.createElement(container, 'div');
+        label.style.marginTop = '1em';
+        label.innerText = this.loading.text ?? 'Loading data...';
+
+        const animationStyles = this.createElement(container, 'style');
+        animationStyles.innerText = [
+            '@keyframes ag-charts-loading { from { opacity: 0 } to { opacity: 1 } }',
+            '@keyframes ag-charts-loading-spinner { to { transform: rotate(0.5turn); } }',
+        ].join('');
```

Comment:
This uses css animations rather than the animation manager as the overlay api requires a `string` not `HTMLElement`.

Reply:
SGTM, let's make use of CSS to do the heavy-lifting where possible.